doc/user/logging.md

```diff
@@ -60,6 +60,54 @@ $ go run main.go --zap-level=1
 {"level":"info","ts":1559866310.065119,"logger":"scoped","msg":"Printing at INFO level"}
 {"level":"debug","ts":1559866310.065123,"logger":"scoped","msg":"Printing at DEBUG level"}
 ```
+## Custom zap logger
+
+In order to use a custom zap logger, [`zapr`][repo_zapr] can be utilized to wrap it in a logr implementation.
+
+Below is an example illustrating the use of [`zap-logfmt`][logfmt_repo] in logging. Here, the `NewLogger` function produces a new `logr.Logger` from an existing `zap.Logger`.
+
+### Example
+
+```Go
+package main
+
+import (
+	"github.com/go-logr/zapr"
+	zaplogfmt "github.com/sykesm/zap-logfmt"
+	"go.uber.org/zap/zapcore"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log logr.Logger
+
+func main() {
+	configLog := zap.NewProductionEncoderConfig()
+	configLog.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
+		encoder.AppendString(ts.UTC().Format(time.RFC3339))
+	}
+
+	logger := zap.New(zapcore.NewCore(
+		zaplogfmt.NewEncoder(configLog),
+		os.Stdout,
+		zapcore.DebugLevel,
+	))
```

Comment:
Should not it be something such as following?

```suggestion
	logger := zap.New(zap.UseDevMode(true))
```
See kubebuider: https://github.com/kubernetes-sigs/kubebuilder/pull/1385/files
c/c @hasbro17

Reply:
I have followed an implementation of zaplogfmt from here - https://godoc.org/github.com/sykesm/zap-logfmt#example-package--Usage

Reply:
Right, so since we're not really able to use the SDK's [`pkg/log/zap`](https://github.com/operator-framework/operator-sdk/blob/master/pkg/log/zap/logger.go) for this example to instantiate a new zapr based logger (since that doesn't let you override the encoder or the format), we might as well use the controller-runtime's [`pkg/log/zap`](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/log/zap/zap.go) pkg and use its [zap.New()](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/log/zap/zap.go#L35) to instantiate a zapr based logger and pass in our own zaplogfmt  encoder.
That should be a little cleaner with functional options and more in line with upstream conventions.

So @varshaprasad96 I think this would make the example something like this:
```Go
import (
	"github.com/go-logr/zapr"
	zaplogfmt "github.com/sykesm/zap-logfmt"
	"go.uber.org/zap/zapcore"
	"go.uber.org/zap"
	logf "sigs.k8s.io/controller-runtime/pkg/log"
        zapcr "sigs.k8s.io/controller-runtime/pkg/log/zap"
)

var log logr.Logger

func main() {
	configLog := zap.NewProductionEncoderConfig()
	configLog.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
		encoder.AppendString(ts.UTC().Format(time.RFC3339))
	}
	logfmtEncoder := zaplogfmt.NewEncoder(configLog)
	logger := zapcr.New(zapcr.UseDevMode(true), zapcr.WriteTo(os.Stdout), zapcr.Encoder(logfmtEncoder))

...

}
```

Reply:
@hasbro17 @camilamacedo86 That makes sense. Thanks! I have made the necessary changes. Please have a look.